fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java

```diff
@@ -0,0 +1,243 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.catalog;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.starrocks.analysis.BrokerDesc;
+import com.starrocks.analysis.DescriptorTable;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.UserException;
+import com.starrocks.fs.HdfsUtil;
+import com.starrocks.proto.PGetFileSchemaResult;
+import com.starrocks.proto.PScalarType;
+import com.starrocks.proto.PSlotDescriptor;
+import com.starrocks.proto.PTypeNode;
+import com.starrocks.rpc.BackendServiceClient;
+import com.starrocks.rpc.PGetFileSchemaRequest;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.ast.ImportColumnDesc;
+import com.starrocks.system.Backend;
+import com.starrocks.thrift.TBrokerFileStatus;
+import com.starrocks.thrift.TBrokerRangeDesc;
+import com.starrocks.thrift.TBrokerScanRange;
+import com.starrocks.thrift.TBrokerScanRangeParams;
+import com.starrocks.thrift.TColumn;
+import com.starrocks.thrift.TFileFormatType;
+import com.starrocks.thrift.TFileType;
+import com.starrocks.thrift.TGetFileSchemaRequest;
+import com.starrocks.thrift.THdfsProperties;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TPrimitiveType;
+import com.starrocks.thrift.TScanRange;
+import com.starrocks.thrift.TTableDescriptor;
+import com.starrocks.thrift.TTableFunctionTable;
+import com.starrocks.thrift.TTableType;
+import org.apache.thrift.TException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+
+public class TableFunctionTable extends Table {
```

Comment:
If the file format is a CSV, what's the internal schema of CSV? `$1, $2 ...`?

Reply:
1. The CSV format is not supported in this PR yet. It would be talked about in the future.
2. Yes.
3. Now, the predicate is not pushed down to the scanner, which could be done in the future.